langserver/references.go

```diff
@@ -75,6 +74,12 @@ func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn jso
 		return PathHasPrefix(path, h.init.RootImportPath)
 	}
 
+	// findRefCtx is used in the findReferences function. It has its own
+	// context so we can stop finding references once we have reached our
+	// limit.
+	findRefCtx, stop := context.WithCancel(ctx)
```

Comment:
just curious, is there a reason why we wouldn't want to just shadow ctx at the beginning of the function to avoid accidentally sending the wrong ctx later?